Engine/Component/ComponentParticleSystem.cpp

```diff
@@ -0,0 +1,338 @@
+#include "ComponentParticleSystem.h"
+
+#include "Main/Application.h"
+#include "Module/ModuleRender.h"
+#include "Module/ModuleProgram.h"
+
+#include "Component/ComponentBillboard.h"
+#include "Module/ModuleResourceManager.h"
+#include "GL/glew.h"
+
+ComponentParticleSystem::ComponentParticleSystem() : Component(nullptr, ComponentType::PARTICLE_SYSTEM)
+{
+	Init();
+}
+
+ComponentParticleSystem::ComponentParticleSystem(GameObject * owner) : Component(owner, ComponentType::PARTICLE_SYSTEM)
+{
+	
+}
+
+void ComponentParticleSystem::Init() 
+{
+	particles.reserve(max_particles);
+
+	billboard = new ComponentBillboard(owner);
```

Comment:
One thing is confusing me: You have the components billboard registered in the ModuleRender class, but when creating here a componentBillboard this one is not being registered right? In case you don't want ti to be registered you are missing to delete de pointer in the componentParticleSystem

Reply:
What do you mean that I don't want it to have it registered?
In this case I create a ComponentBillboard for the whole componentParticleSystem(PS), so there is only one texture for the PS. The componentBillboard is a separe thing from the PS, so in the moduleRender, we create a vector of Billboards in case someone wants to create a billboard and not a PS.
But you are right in the part of deleting, I totally forgot to do it. But I don't understand the part of "You have the components billboard registered in the ModuleRender class, but when creating here a componentBillboard this one is not being registered right?"
Sorry :/

Reply:
Ok I saw you added the delete, with that is ok. I will test the branch this evening and approve if it is working fine